tests/test_diffraction_objects.py

```diff
@@ -236,18 +236,21 @@ def test_dump(tmp_path, mocker):
     directory = Path(tmp_path)
     file = directory / "testfile"
     test = Diffraction_object()
-    test.wavelength = 1.54
+    test.wavelength = 1.01
     test.name = "test"
     test.scat_quantity = "x-ray"
     test.insert_scattering_quantity(
         x, y, "q", metadata={"thing1": 1, "thing2": "thing2", "package_info": {"package2": "3.4.5"}}
     )
-    with mocker.patch("importlib.metadata.version", return_value="3.3.0"), freeze_time("2012-01-14"):
+
+    mocker.patch("importlib.metadata.version", return_value="3.3.0")
```

Comment:
The second warning is resolved by not using `with`  because the mocker fixture automatically handles the setup and teardown of mocks without needing manual intervention.